File: api/routers/agents.py
from __future__ import annotations

from fastapi import APIRouter, Depends, Request

from api.auth import verify_api_key
from api.models.schemas import AgentInfo


router = APIRouter(
    prefix="/agents", tags=["agents"], dependencies=[Depends(verify_api_key)]
)


@router.get("", response_model=list[AgentInfo])
async def list_agents(request: Request):
    coordinator = request.app.state.state.coordinator
    agents = await coordinator.list_agents()
    # Best-effort: only return a subset of fields.
    return [AgentInfo(**a) for a in agents]

File: packages/api/src/routes/agents.ts
import type { FastifyInstance } from "fastify";
import { verifyApiKey } from "../middleware/auth.js";
import { AgentInfo } from "../schemas.js";
import type { AppState } from "../services/state.js";

export async function agentRoutes(app: FastifyInstance): Promise<void> {
  app.addHook("preHandler", verifyApiKey);

  app.get("/agents", async () => {
    const state = (app as unknown as { state: AppState }).state;
    const agents = await state.coordinator.listAgents();
    return agents.map((a) => AgentInfo.parse(a));
  });
}
